FRW/Services/Manager/WalletManager.swift

```diff
@@ -796,6 +800,41 @@ extension WalletManager {
         }
     }
 
+    func fetchAccountInfo() async throws {
+        do {
+            let accountInfo = try await FlowNetwork.checkAccountInfo()
+            await MainActor.run {
+                self.accountInfo = accountInfo
+            }
+        } catch let error {
+            log.error("[EVM] fetch account info failed.\(error.localizedDescription)")
+            throw error
+        }
+    }
+    
+    var minimumStorageBalance: Decimal {
+        guard let accountInfo else { return Self.minDefaultBlance }
+        return accountInfo.storageFlow * Self.minDefaultBlance;
+    }
+    
+    var isStorageInsufficient: Bool {
+        return isStorageInsufficient(for: 0)
+    }
+
+    func isStorageInsufficient(for amount: Decimal) -> Bool {
+        guard let accountInfo else { return false }
+        return accountInfo.availableBalance - amount < Self.minDefaultBlance
+    }
+    
+    var isBalanceInsufficient: Bool {
+        return isBalanceInsufficient(for: 0)
+    }
+    
+    func isBalanceInsufficient(for amount: Decimal) -> Bool {
+        guard let accountInfo else { return false }
+        return accountInfo.availableBalance - amount - Self.minDefaultBlance < Self.minDefaultBlance
```

Comment:
I assume the `- Self.minDefaultBlance` is the moveFee ?

Reply:
Correct — I've renamed all constants using the names you provided in your pseudo-code.